heka/sandbox/encoders/http_edge_encoder.lua

```diff
@@ -68,24 +53,25 @@ local ns_config = {
 }
 
 function process_message()
-    -- Carry forward payload some incoming fields
+    -- Carry forward payload some incoming fields.
     msg.Payload = read_message("Payload")
     msg.Timestamp = read_message("Timestamp")
     msg.EnvVersion = read_message("EnvVersion")
 
-    -- Hostname is the host name of the server that received the message
+    -- Hostname is the host name of the server that received the message.
     msg.Hostname = read_message("Hostname")
 
     -- Host is the name of the HTTP endpoint the client used (such as
-    -- "incoming.telemetry.mozilla.org")
+    -- "incoming.telemetry.mozilla.org").
     msg.Fields.Host = read_message("Fields[Host]")
 
-    -- Path should be of the form ^/submit/namespace/id[/extra/path/components]$
+    -- Path should be of the form:
+    --     ^/submit/namespace/id[/extra/path/components]$
     local path = read_message("Fields[Path]")
-    local components = split_path(path)
+    local components = split_path(path, "/")
 
     -- Skip this message: Not enough path components.
-    if components == nil or table.getn(components) < 3 then
+    if not components or #components < 3 then
```

Comment:
components can still never be nil from the current split function (i.e., the test is uncessary)